test/jit/test_save_load.py

```diff
@@ -20,6 +20,54 @@
 
 
 class TestSaveLoad(JitTestCase):
+    def test_versioned_symbols(self):
+        """
+        Tests Torchscript symbol versioning. See note [Versioned Symbols].
+        This test uses an undocumented, test-only function torch._subcmul.
+        This function is implemented as (a - alpha * b) with a default value
+        of 1 for alpha. In file format version 2, however, it was implemented
+        as (b - alpha * a) with a default value of 2 for alpha.
+        This test verifies a module seralized with file format version 2
+        exhibits the old behavior, and that the same module newly serialized
+        exhibits the current behavior.
+        #
+        """
+        class MyModule(torch.nn.Module):
+            def __init__(self):
+                super(MyModule, self).__init__()
+
+            def forward(self, a, b, alpha: float):
+                no_alpha = torch._subcmul(a, b)
+                with_alpha = torch._subcmul(a, b, alpha)
+                return no_alpha, with_alpha
+
+        def historic_subcmul(a, b, alpha=2):
+            return b - alpha * a
+
+        def current_subcmul(a, b, alpha=1):
+            return a - alpha * b
+
+        # Loads and verifies the historic behavior of the module
+        # that was serialized with version 2
+        module_v2 = torch.jit.load(pytorch_test_dir + "/jit/fixtures/subcmul_v2.pt")
+        a = torch.randn((5,))
+        b = torch.randn((5,))
+        alpha = random.random()
+        args = (a, b, alpha)
+        no_alpha_v2, with_alpha_v2 = module_v2(*args)
+        self.assertEqual(no_alpha_v2, historic_subcmul(a, b))
+        self.assertEqual(with_alpha_v2, historic_subcmul(*args))
+
+        # Scripts, saves, loads and verifies the current behavior of the module
```

Comment:
this is a useful test, but I guess I was also expecting an "upgrader" test.  That is, I do re-serialize module_v2 (not the currentMyModule()) and somehow check that I don't serialize the old symbol/version.  I'm not sure how that would actually look in practice, though.

Reply:
Good point. I can extend the test to include that check.